### DISCUSSION TITLE
unable mpiexec remote computer

### DISCUSSION THREAD

**Discussion by sunho0248** [firemodels/fds#12590] [Q&A] [answered]
FDS-6.8.0

when i mpi test at my local(SJ046) computer is ok
but remote computer(SJ189) can't do it
SJ046 -> SJ189
 
hosts.txt)
SJ046:2
SJ189:2
--------------------------------------------------------------------------------------------------------
C:\Users\SJ046\Desktop\Sim\Test>mpiexec -machine hosts.txt test_mpi
[mpiexec@SJ046] HYDI_bstrap_service_launch (bstrap\service\service_launch.c:432): server rejected credentials
[mpiexec@SJ046] remote_launch (bstrap\src\intel\i_hydra_bstrap.c:607): error launching bstrap proxy
[mpiexec@SJ046] single_launch (bstrap\src\intel\i_hydra_bstrap.c:665): remote launch error
[mpiexec@SJ046] launch_bstrap_proxies (bstrap\src\intel\i_hydra_bstrap.c:849): single launch error
[mpiexec@SJ046] HYD_bstrap_setup (bstrap\src\intel\i_hydra_bstrap.c:1043): unable to launch bstrap proxy
[mpiexec@SJ046] HYD_print_bstrap_setup_error_message (intel\i_mpiexec.c:1027): error setting up the bootstrap proxies
[mpiexec@SJ046] Possible reasons:
[mpiexec@SJ046] 1. Host is unavailable. Please check that all hosts are available.
[mpiexec@SJ046] 2. Cannot launch hydra_bstrap_proxy.exe or it crashed on one of the hosts. Make sure hydra_bstrap_proxy.exe is available on all hosts and it has right permissions.
[mpiexec@SJ046] 3. Firewall refused connection. Check that enough ports are allowed in the firewall and specify them with the I_MPI_PORT_RANGE variable.
[mpiexec@SJ046] 4. service bootstrap cannot launch processes on remote host. You may try using -bootstrap option to select alternative launcher.

 

remote computer same mpi and
regist&validate done

my computer network is PAT

window 10 Home edition

sorry for my english

**Comment by mcgratta** [comment]
I do not think this is going to work. PAT is Port Address Translation, which allows multiple computers to use the same IP address. I would think that the MPI messaging will not work with this. Plus, you have Windows 10 Home edition which may not have all the necessary features to run an MPI job across the network.

**Reply by sunho0248** [reply]
I tried other computers win 10 pro
in same network.
A computer IPv4 ip=192.168.0.21
B computer IPv4 ip=192.168.0.23

However, it still showed the same error message as I showed. Is there any way to run MPI on the PAT network?

**Reply by drjfloyd** [reply]
PAT is for the public facing IP address and not the LAN addresses.  The addresses given (192.168.x) are internal LAN addresses and are different. PAT may not be the problem. The problem may be this:

"Firewall refused connection. Check that enough ports are allowed in the firewall and specify them with the I_MPI_PORT_RANGE variable."

This could be either the computer firewall or at the network switch.  If you have an IT department they may be able to help diagnose if this is the issue.

**Reply by sunho0248** [reply]
@drjfloyd 
Do I need to assign the I_MPI_PORT_RANGE variable separately?

I've applied the same rules for FDS as Pyrosim based on the firewall inbound rules.

**Reply by sunho0248** [reply]
Could you teach me how to set up a firewall for MPI usage?

**Reply by msimmar26** [reply]
I had the same problem and what I needed was to first ask my IT department to open the specific ports, and then assign them in the I_MPI_PORT_RANGE, as a separate line. I'll explain my case here, because I've also had some problems.

**Comment by msimmar26** [comment]
In my case I'm struggling with mpiexec and I wonder if there is a limit of computers or CPUs per computers to be used.
Both mpi_test and a simple simulation work when I invoke just 1 process per computer in my hosts.txt file.

Example of hosts file for 3 MPI processes: 
XXX.XX.XXX.XX1 : 1
XXX.XX.XXX.XX2 : 1
XXX.XX.XXX.XX3 : 1

But when I set more than 1 MPI process per computer the process shows "Starting FDS..." but never carries on.
XXX.XX.XXX.XX1 : 1
XXX.XX.XXX.XX2 : 2

Here are the commands I'm using:
**>mpiexec -register**
User: xxxx
password: xxxx

**>set I_MPI_PORT_RANGE=54912:54912**  

**>mpiexec -wdir \Marina\Test_CFG2\Prova_1 -machine \Marina\Test_CFG2\Prova_1\hosts.txt test_mpi**

**>mpiexec -wdir \Marina\Test_CFG2\Prova_1 -machine \Marina\Test_CFG2\Prova_1\hosts.txt fds \Marina\Test_CFG2\Prova_1\room3.fds**

I'm working with a set of 20 computers all of them connected within the same Windows network (my University network). One of the ports has been opened previously to avoid the Firewall action (54912). The simulations are run in the same shared working directory.

Here are the specifications of the computers I'm using (in case the problem is derived from the PCs):

Intel(R) Core(TM) i5-6400 CPU @ 2.70GHz
	Base speed: 2.70 GHz
	Sockets: 1
	Cores: 4
	Logical processors: 4
	Virtualization: Enabled
	L1 cache: 256 kB
	L2 cache: 1.0 MB
	L3 cache: 6.0 MB

And this is the script for the simple simulation:

&HEAD CHID='room3' TITLE='One room garden house example internet' /

&MESH IJK=15,30,20  XB=0,1.5,0,3,0,2, MPI_PROCESS=0 / 3m x 3m x 2m
&MESH IJK=15,30,20  XB=1.5,3,0,3,0,2, MPI_PROCESS=1 / 3m x 3m x 2m
&MESH IJK=30,30,20  XB=0,3,0,3,2,4, MPI_PROCESS=2 / 3m x 3m x 2m

&MISC SHARED_FILE_SYSTEM = .TRUE. /

&TIME T_END=30./

&DUMP NFRAMES=30/

&VENT XB= 0,0,0.5,2.5,0,1.5,  SURF_ID='OPEN'/ PORTA
&VENT XB= 0.5,1.5,0,0,1,1.5,  SURF_ID='OPEN'/ FINESTRA
&VENT XB= 0.5,1.5,3,3,1,1.5,  SURF_ID='OPEN'/ FINESTRA

&SURF ID='fire', HRRPUA=125 /
&OBST XB= 2.5,3,2.5,3,0,0.2 SURF_ID='fire' /

&REAC FUEL='ACETONE' /

&SLCF PBY=2.75, QUANTITY='TEMPERATURE' / 

&SLCF PBX=2.75, QUANTITY='TEMPERATURE'

&DEVC XBP=2.75,2.75,2.75,2.75,0.2,2, QUANTITY='TEMPERATURE', ID='THERMOCOUPLES', POINTS=9, POINTS_ARRAY_Z=0.2,0.4,0.6,0.8,1,1.2,1.4,1.6,1.8 /

&TAIL/



So, what do you think the problem could be?

**Reply by sunho0248** [reply]
Retry with delete "MPI_PROCESS=0, MPI_PROCESS=1, MPI_PROCESS=2"

**Reply by msimmar26** [reply]
I did it and didn't work either.

**Comment by mcgratta** [comment]
Will this job run if you assign each mesh to a different computer?

**Reply by msimmar26** [reply]
If I assign 1 mesh to a different computer it works as long as I specify 1 MPI process per computer in my hosts file. However, if I specify more than 1 mpi processes per computer in the mentioned file, the simulation never advances. Don't know if you mean that...

**Comment by sunho0248** [comment, marked as answer]
Sorry, my mistake was not creating the same account on the computer.
It's been resolved. Thank you all for your help.
